### DISCUSSION TITLE
TARGET=nrf does not initializes node_id properly

### DISCUSSION THREAD

**Discussion by rahav** [contiki-ng/contiki-ng#2763] [Q&A] [answered]
I now fully switched from using the `nrf52840` target to `nrf`. I am using the `deployment` module in my research. When the same code that works with `nrf52840` is compiled with `nrf` all nodes received `node_id = 0`.

Any suggestions?

**Comment by pjonsson** [comment, marked as answer]
The nrf platform uses some API from nrfx for getting the device address. Did the `linkaddr_node_addr` change so the deployment table needs to be updated?

**Reply by rahav** [reply]
As always, you are very insightful. From the bottom of my heart, a big THANK YOU!
I'll go on record and say that if you ever need help, bringing the moon down, or hauling sofas, you can contact me.

The `linkaddr_node_addr` indeed changes. I will change my mapping table because I really want to move to the newer `nrf`.

Long term, what is the plan? Every `TARGET` will require a new mapping? If my opinion matters, it shouldn't. But I wonder if there a bootstrapping challenge here, to make the `deployment` module really a mapping from a fixed MAC-addr to node_id?

**Reply by pjonsson** [reply]
Maintaining a target takes a bit of work, so the long term plan is to only have one target for each hardware platform. I think this will also eliminate the issue you ran into here.
